# Supervised Learning Concepts & the Scikit-learn API

## Hands-On Lab: Setting Up the ML Workflow

In [23]:
import pandas as pd
from sklearn.model_selection import train_test_split

## Step 1: Load the Dataset

The Titanic dataset was loaded using **pandas**.

The dataset contains **891 rows** and **12 columns**, confirming that it was loaded successfully and is ready for preprocessing and model training.

In [24]:
df = pd.read_csv("Titanic-Dataset.csv")
df.shape

(891, 12)

## Step 2: Separate Features and Target

The **Survived** column was selected as the target variable (`y`), while all remaining columns were assigned to the feature set (`X`).

In [ ]:
X = df.drop("Survived", axis=1)
y =df["Survived"]

## Step 3: Split the Dataset into Training and Testing Sets

The dataset was divided into **80% training data** and **20% testing data** using `train_test_split()`.

A fixed `random_state` was used to ensure that the split is reproducible.

In [19]:
X_train , X_test , y_train , y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)

## Step 4: Verify the Shapes of the Datasets

The shapes of the training and testing datasets were printed to confirm that the split was successful and that the number of samples is consistent between the features and target variables.

In [20]:
print("X_train:", X_train.shape)
print("X_test :", X_test.shape)
print("y_train:", y_train.shape)
print("y_test :", y_test.shape)

X_train: (712, 11)
X_test : (179, 11)
y_train: (712,)
y_test : (179,)


### Result

The dataset was successfully split into training and testing sets using an **80/20** ratio.

- **Training set:** 712 samples
- **Testing set:** 179 samples

The number of samples in the feature matrices matches the corresponding target vectors, confirming that the split was performed correctly.

## Why Should the Model Never See the Test Set During Training?

The test set must remain unseen during training because it is used to evaluate the model on new, unseen data.

If the model learns from the test set, the evaluation becomes biased and no longer reflects the model's true ability to generalize. Keeping the test set separate provides a fair and reliable estimate of real-world performance.